wgengine/tstun/tun.go

```diff
@@ -129,6 +131,14 @@ func WrapTUN(logf logger.Logf, tdev tun.Device) *TUN {
 	return tun
 }
 
+// SetDestIPActivityFuncs sets a map of funcs to run per packet
+// destination (the map keys).
+//
+// The map ownership passes to the TUN. It must be non-nil.
+func (t *TUN) SetDestIPActivityFuncs(m map[packet.IP]func()) {
```

Comment:
Metacomment: not a huge fan of these Set* methods. The fact that we can't provide all this at construction time is further evidence that these packages are super entangled with each other, and makes it significantly harder to follow wtf the code path actually is.

Not asking you to do anything about it, just noting that these kinds of setters are evidence of a refactor we need at some point.

Reply:
I get your frustration, but also: constructors shouldn't require optional parameters.

But I guess we could do an ... option pattern thing.

Reply:
I was thinking more of NewBlah(&Options{...}), with fields that document that they can be non-nil. We do that in various places already.

Also fine with ...Option, but we should pick one or the other, and I think I prefer (&Options{...}) as being a bit less magical and varargs-ey.

Reply:
In any case, that wouldn't help here. I need to swap out the entire map at runtime regularly (when wireguard-go is reprogrammed and there's a new set of IPs to track the activity of).